R/logRatioPlot.R

```diff
@@ -1,253 +1,565 @@
-
-    assertthat::assert_that(plotType %in% c("bar", "point"),
-                            msg = "plotType must be either 'bar' or 'point'.")
-
-    .addGeoms <- function(myPlot){
-        if (tolower(plotType) == "bar") {
-            myPlot <- myPlot + geom_bar(stat = "identity",
-                                        alpha = barAlpha,
-                                        color = barColor,
-                                        fill = barFill,
-                                        size = barSize,
-                                        width = barWidth)
-        } else if (tolower(plotType) == "point") {
-            myPlot <- myPlot + geom_point(alpha = pointAlpha,
-                                          color = pointColor,
-                                          fill = pointFill,
-                                          size = pointSize,
-                                          shape = pointShape)
+                         labelAngle = 45,
+                         axisFree = TRUE) {
+    assertthat::assert_that(!missing(dgeObj),
+                            !is.null(dgeObj),
+                            "DGEobj" %in% class(dgeObj),
+                            msg = "dgeObj must be specified and should be of class DGEobj")
+
+    topTables <- DGEobj::getType(dgeObj, "topTable")
+    assertthat::assert_that(length(topTables) > 0,
+                            msg = "No topTable dataframes found in dgeObj. Please specify a dgeObj that contains topTable dataframes.")
+
+    assertthat::assert_that(all(sapply(topTables, class) == "data.frame"),
+                            msg = "topTables list must only contain dataframes.")
+
+    minNameLen <- min(sapply(names(topTables), nchar))
+    assertthat::assert_that(minNameLen > 0,
+                            msg = "All dataframes in topTable list must be named (it must be a named list.)")
+
+    geneData <- DGEobj::getItem(dgeObj, "geneData")
+    assertthat::assert_that(!is.null(geneData),
+                            nrow(geneData) > 0,
+                            msg = "No geneData is found in dgeObj. Please specify a dgeObj that contains geneData.")
+
+    plotType <- tolower(plotType)
+    if (any(length(plotType) != 1,
+            !plotType %in% c("canvasxpress", "ggplot"))) {
+        warning("plotType must be either canvasXpress or ggplot. Setting default value 'canvasXpress'")
+        plotType <- "canvasxpress"
+    }
+
+    topTables <- lapply(topTables, function(x) {
+        x$EnsgID <- rownames(x)
+        rownames(x) <- NULL
+        x})
+
+    for (name in names(topTables)) {
+        topTables[[name]]["Contrast"] <- name
+    }
+
+    contrastsDF <- do.call(rbind, topTables)
+
+    # Add gene symbols from geneData
+    genesSymbols <- data.frame("EnsgID" = row.names(geneData), GeneSymbol = geneData$rgd_symbol)
+    contrastsDF <-  dplyr::left_join(contrastsDF, genesSymbols, by = "EnsgID")
+
+    assertthat::assert_that(!missing(facetColname),
```

Comment:
Can't you have default values for facetColname & xcolName.

Reply:
done